components/plan-build/bin/hab-plan-build.sh

```diff
@@ -961,8 +961,8 @@ _resolve_scaffolding_dependencies() {
     scaff_build_deps_resolved+=("$resolved")
     # Add each (fully qualified) direct run dependency of the scaffolding
     # package.
-    read -r -a sdeps <<< "$(_get_deps_for "$resolved")"
-    for sdep in "${sdeps[@]}"; do
+    sdeps="$(_get_deps_for "$resolved")"
+    for sdep in $sdeps; do
```

Comment:
This was [correct before](https://github.com/habitat-sh/habitat/blob/d0b63fdc6cc6942bbe308e05c368c1045315f16d/components/plan-build/bin/hab-plan-build.sh#L966-L967), and somehow I think we got mined up as to whether we had input that was IFS-delimited or line delimited. What we want is:
```bash
    mapfile -t sdeps < <(_get_deps_for "$resolved")
    for sdep in "${sdeps[@]}"; do
```

Similarly for the other places.

See https://github.com/koalaman/shellcheck/wiki/SC2207#prefer-mapfile-or-read--a-to-split-command-output-or-quote-to-avoid-splitting